library/general/src/lib/ui/text_helpers.rb

```diff
@@ -20,6 +20,59 @@
 module UI
   # Provides a set of methods to manipulate and transform UI text
   module TextHelpers
+    # Default HTML tags replacements
+    #
+    # @see #plain_text
+    # @return [Hash<String, String>]
+    DEFAULT_HTML_TAGS_REPLACEMENTS = {
+      "<br>"  => "\n",
+      "<br/>" => "\n",
+      "<br >" => "\n",
+      "</li>" => "\n",
+      "<ol>"  => "\n\n",
+      "<ul>"  => "\n\n",
+      "</p>"  => "\n\n"
+    }.freeze
+
+    # Get a new text version after ridding of HTML tags
+    #
+    # By default, a fully plain text will be returned, since some tags (see
+    # {DEFAULT_HTML_TAGS_REPLACEMENTS}) are going to be replaced by line breaks while the rest of
+    # them will be removed.
+    #
+    # However, the _tags_ param allows specifying which tags should be replaced
+    # or removed.  Additionally, a collection of tag => replacement can be
+    # provided via the _replacements_ param.
+    #
+    # Note that all matched tags without a replacement are going to be replaced
+    # with `nil`. In other words, they will be deleted.
+    #
+    # @example Remove all HTML tags
+    #   text = "<p>YaST:</p><p>a <b>powerful</b> installation and configuration tool.</p>"
+    #   plain_text(text) #=> "YaST:\n\na powerful installation and configuration tool."
+    #
+    # @example Removing only the <p> tags
+    #   text = "<p>YaST:</p><p>a <b>powerful</b> installation and configuration tool.</p>"
+    #   plain_text(text, tags: ["p"])
+    #     #=> "YaST:\n\na <b>powerful</b> installation and configuration tool."
+    #
+    # @example Using custom replacements
+    #   text = "<p>YaST:</p><p>a <b>powerful</b> installation and configuration tool.</p>"
+    #   plain_text(text, replacements: { "<p>" => "\n- ", "<b>" => "*", "</b>" => "*" })
+    #     #=> "- YaST:\n- a *powerful* installation and configuration tool."
+    #
+    # @param text [String] text to be processed
+    # @param tags [Array<String>] specific tags to be replaced or deleted
+    # @param replacements [Hash<String, String>] the replacements collection, tag => replacement
```

Comment:
what about using block version instead of hash parameter? it is more flexible. so then you have something like
```
block ? block.call(match.to_s.downcase) : replacements[match.to_s.downcase]
```

Reply:
so it is more like gsub API and definitivelly more flexible one if you want to use some kind of state machine like replacing li by number or unicode dot depending if it is ol or ul.

Reply:
Sorry, I'm not sure if I got what you mean.

Could you please elaborate it a little bit more?

Reply:
well, what I mean is that instead of having replacements as hash parameter, you can accept block in same way like gsub allows it ( or allow both same as gsub ).
so something like
```
def plain_text(text, tags: nil, replacements: nil, &block)
....
text.gsub(regex) do |match|
  block ? block.call(match.to_s.downcase) : replacements[match.to_s.downcase]
end
```

This way, you can have something like
```
  text.plain_text do |param|
    case param
    when "<ol>" 
      @ordered = true
       index = 0
    when "<ul>" then @ordered = false
    when "<li>" then @ordered ? "#{index += 1}. " : "•"
    end
  end
```

which also create nicer plain text lists.